openvpn/session/authentication.go

```diff
@@ -0,0 +1,39 @@
+package session
+
+import (
+	"github.com/mysterium/node/identity"
+	client_auth "github.com/mysterium/node/openvpn/middlewares/client/auth"
+	server_auth "github.com/mysterium/node/openvpn/middlewares/server/auth"
+	"github.com/mysterium/node/session"
+)
+
+const sessionSignaturePrefix = "MystVpnSessionId:"
+
+// SignatureCredentialsProvider returns session id as username and id signed with given signer as password
+func SignatureCredentialsProvider(id session.SessionID, signer identity.Signer) client_auth.CredentialsProvider {
+	return func() (string, string, error) {
```

Comment:
I'm not fully aware, what do we gain from returning function here instead of credentials itself:
- The on-demand aspect doesn't really look useful here, because we're not gaining much.
- We doesn't seem to be mocking this anywhere.

Reply:
That was designed with other PR. Might be one of the places for future refactoring